pest/src/iterators/flat_pairs.rs

```diff
@@ -102,6 +102,22 @@ impl<'i, R: RuleType> FlatPairs<'i, R> {
     }
 }
 
+impl<'i, R: RuleType> ExactSizeIterator for FlatPairs<'i, R> {
+    fn len(&self) -> usize {
+        let mut start = self.start;
+        let mut count = 0;
+        while start < self.end {
+            start += 1;
+            while start < self.end && !self.is_start(start) {
+                start += 1;
+            }
+
+            count += 1;
+        }
+        count
+    }
```

Comment:
is it ok for a `len` to have a linear complexity / a loop there? maybe we can check with this benchmark https://github.com/pest-parser/pest/blob/master/grammars/benches/json.rs#L91 to see if it had any impact? (@huacnlee used to to check that the impact of the line index )
alternatively, perhaps there could be a private field to maintain the length iteratively?